src/app/clusters/level-control/level-control.cpp

```diff
@@ -240,7 +240,7 @@ class DefaultLevelControlSceneHandler : public scenes::DefaultSceneHandlerImpl
         // TODO : Implement action on frequency when frequency not provisional anymore
         // if(LevelControlHasFeature(endpoint, LevelControl::Feature::kFrequency)){}
 
-        if (!chip::app::NumericAttributeTraits<uint8_t>::IsNullValue(level))
+        if (!app::NumericAttributeTraits<uint8_t>::IsNullValue(level))
```

Comment:
This doesn't seem to match spec.  Spec says that anything outside [MinLevel, MaxLevel] should be treated as null, no?

Followup is best for fixing this, but please don't lose track of it.

Reply:
The check as described in spec happens on the AddScene command (implementation ongoing here: https://github.com/project-chip/connectedhomeip/pull/32632), converting any values outside bounds to NULL. Therefore, on a RecallScene that would call this handler, the value should either valid or null at this point no?

Reply:
So the thing is, what happens if a scene is stored, then OTA happens and Min/MaxLevel change and make the stored scene invalid?

It sounds like the spec needs to make it much more clear how this situation is handled for scenes in general.

Reply:
Since we are using the same helper function than the MoveToLevel command `moveToLevelHandler` I assumed the level control cluster would be able to handle it the same way it handles any commands outside of its bounds. 

The idea was that this section of the code should handle it:
```
    if (state->maxLevel <= level)
    {
        state->moveToLevel = state->maxLevel;
    }
    else if (level <= state->minLevel)
    {
        state->moveToLevel = state->minLevel;
    }
    else
    {
        state->moveToLevel = level;
    }
```
But now I see that the this clashes with the spec, the ExtensionFieldSet behavior states this should behave like null, whereas the level control cluster constrains the value to the closest maximum since it assumes a uint8 in the MoveToLevel command.

I added a check in the handler, this will be something to take in consideration in the other PR as well, we cannot only rely on storage to catch out of bounds values